README.md

```diff
@@ -224,13 +123,136 @@ You'll just have to make sure that your app doesn't have any tweens that do this
 Tweens should encode any redirect locations that they generate,
 [like this](https://github.com/hypothesis/pyramid-sanity/blob/d8492620225ec6be0ba28b3eb49d329ef1e11dc2/src/pyramid_sanity/_egress.py#L22-L30).
 
+Error details
+-------------
+
+If you would like to reproduce the errors an [example app](#addendum-example-application) is given at the end
+of this section. All of the presented `curl` commands work with this app.
+
+### Badly encoded query parameters makes `request.GET` crash
+
+```terminal
+curl 'http://localhost:6543/foo?q=%FC'
+```
+
+**By default**
+
+WebOb raises `UnicodeDecodeError`. As there is no built-in exception view for
+this exception the app crashes.
+
+**With `pyramid-sanity`**
+
+A `pyramid_sanity.exceptions.InvalidQueryString` is returned which results in a
+`400: Bad Request` response.
+
+Related issues:
+
+* https://github.com/Pylons/pyramid/issues/3399
+* https://github.com/Pylons/webob/issues/161
+
+### A badly encoded path can cause a crash
+
+```terminal
+curl 'http://localhost:6543/%FC'
+```
+
+**By default**
+
+Pyramid raises [`pyramid.exceptions.URLDecodeError`](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/exceptions.html#pyramid.exceptions.URLDecodeError).
+As there is no built-in exception view for this exception the app crashes.
+
+**With `pyramid-sanity`**
+
+A `pyramid_sanity.exceptions.InvalidURL` is returned which results in a
+`400: Bad Request` response.
+
+**Related issues**
+
+* https://github.com/Pylons/pyramid/issues/434
+* https://github.com/Pylons/pyramid/issues/1374
+* https://github.com/Pylons/pyramid/issues/2047
+* https://github.com/Pylons/webob/issues/114
+
+### Bad or missing multipart boundary declarations make `request.POST` crash
```

Comment:
More specific about the cause here, it's specifically boundary declarations, rather than bad encoding in general.